### ISSUE TITLE
Flask implementation - Pyctuator messes with app logging conf

### ISSUE THREAD

**Issue by jpays** [SolarEdgeTech/pyctuator#31]
Hi, 

I'm trying to install Pyctuator on our existing app on Flask/uWSGI.

Everything works fine except that pyctuator is modifying the logging configuration of the application, and some things don't work as expected after Pyctuator is set up.

Our app is created the following way (redacted to the relevant part only) :

```python
def create_app():

    """ Flask application factory """

    # create and configure the app
    app = Flask(__name__, instance_path=instance_path, instance_relative_config=True)
    app.config.from_json('config.json')

    # Allow CORS from angers.cnp.fr
    CORS(app)

    # Add proxyfix middleware to handle redirects correctly
    # x_port = 0 to work around Kong sending incorrect X-Forwarded-Port header
    app.wsgi_app = ProxyFix(app.wsgi_app, x_host=1, x_port=0)

    # configure the DB
    # pool_recycle must be set below 3600 to avoid the session to be killed by the firewall
    engine = create_engine(app.config["DATABASE_URI"], convert_unicode=True, pool_pre_ping=True, pool_recycle=3500)

   # Enable pyctuator
    pyctuator = Pyctuator(
        app,
        "Plateforme IA",
        pyctuator_endpoint_url=app.config["ROOT_API_PATH"]+'/actuator',
        app_url=None,
        registration_url=None
    )


    pyctuator.set_build_info(
        name="Plateforme IA",
        version=app.config["VERSION"],
        time=dateutil.parser.parse(app.config["BUILD_DATE"]),
    )

    pyctuator.register_health_provider(DbHealthProvider(engine))

    sentinel = Sentinel(app.config["CACHE_REDIS_SENTINELS"], socket_timeout=1.0)
    master = sentinel.master_for(app.config["CACHE_REDIS_SENTINEL_MASTER"], socket_timeout=1.0)
    pyctuator.register_health_provider(RedisHealthProvider(master))

    return app

application = create_app()
```

* Before setting up Pyctuator, any runtime exception will be logged in uwsgi_error.log and I can use the root logger anywhere in an endpoint code and it gets logged in uwsgi_error.log, such as :
 ```python
import logging
logging.critical("Test")
```
* After setting up Pyctuator, nothing is logged in uwsgi_error.log at all. 

This is not acceptable for us.

Any idea why we have this behaviour ?

FYI if I comment the line 92 in pyctuator.py, the problem goes away :
```python
      root_logger.addHandler(self.pyctuator_impl.logfile.log_messages)
```

Thanks

**Comment by yanoom** [comment]
Hi,
we're happy you're using Pyctuator, and appreciate your effort to write the issue in details...

Of course this is not by-design, and should be fixed!!
:)

We'll update regarding a fix asap

**Comment by jpays** [comment]
Thank you.

I figured out to make a very simple program to reproduce the issue.

* Install flask and pyctuator in a venv

* Create `app.py` as such :
```
import logging
from flask import Flask
from pyctuator.pyctuator import Pyctuator

app = Flask(__name__)

@app.route('/')
def _():
    logging.critical("This should be logged")
    return 'OK'

#pyctuator = Pyctuator( app, "test", pyctuator_endpoint_url='/actuator', app_url=None, registration_url=None)
```

* In the directory execute `flask run`
* go to http://localhost:5000/, you should see a page with "OK" and a message should be logged on the stderr of the flask comand

* Now if you uncomment the pyctuator line and try again, you should see that the message is no longer logged on stderr

**Comment by yanoom** [comment]
👍
I was able to reproduce!!

**Comment by yanoom** [comment]
Looking at your reproducer app:
while the `critical` log entires aren't shown in console, I can see them via the Pycatuator logfile API (`/actuator/logfile` in your case).

The reason is that the python logging system wasn't initialized before Pyctuator started.
In this case Pyctuator's logging-handler (the one that's responsible for intercepting log messages and making them available via the API) is the only handler that is installed...

As you can see in https://docs.python.org/3/library/logging.html#logging.basicConfig:
> The functions debug(), info(), warning(), error() and critical() will call basicConfig() automatically if no handlers are defined for the root logger.

To fix this, you should initialize the logging system before starting Pyctuator by calling any of:
* `logging.info("some message")` 
* `logging.basicConfig(level=logging.INFO)`
(we've used the latter in [flask_example_app](https://github.com/SolarEdgeTech/pyctuator/blob/master/examples/Flask/flask_example_app.py))

Regardless, I will change Pyctuator so it will initialize the logging system if not yet initialized which will solve this issue.

**Comment by yanoom** [comment]
@jpays, please confirm this fix in [v0.13](https://pypi.org/project/pyctuator/)
🙂

**Comment by jpays** [comment]
Ok, this makes perfect sense.

I upgraded to v0.13 and I confirm that the behaviour is fixed in the reproducer app, and also in my real application.

Thanks for your actions :)
